AK/ByteBuffer.h

```diff
@@ -293,20 +257,25 @@ inline void ByteBufferImpl::zero_fill()
 
 inline NonnullRefPtr<ByteBufferImpl> ByteBufferImpl::create_uninitialized(size_t size)
 {
-    return ::adopt_ref(*new ByteBufferImpl(size));
+    auto* buffer = static_cast<ByteBufferImpl*>(kmalloc(sizeof(ByteBufferImpl) + size));
+    VERIFY(buffer);
+    new (buffer) ByteBufferImpl { size };
+    return ::adopt_ref(*buffer);
```

Comment:
This is very odd (casting to a pointer to T before there actually is a T there) could you instead do
```suggestion
auto* buffer = kmalloc(sizeof(ByteBufferImpl) + size);
VERIFY(buffer);
return adopt_ref(*new (buffer) ByteBufferImpl(size));
```

Reply:
Aye, that makes sense. :)